client/components/add-button.jsx

```diff
@@ -1,7 +1,127 @@
 import React from 'react';
 
-export default function AddButton(props) {
-  return (
-    <a className='add-button cursor-pointer' onClick={props.openModal}>Add</a>
-  );
+export default class AddButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      quantity: 0,
+      measurementUnit: '#',
+      foodCategory: 'fruits'
+    };
+    this.updateMeasurement = this.updateMeasurement.bind(this);
+    this.categoryClicked = this.categoryClicked.bind(this);
+    this.submitItem = this.submitItem.bind(this);
+    this.updateName = this.updateName.bind(this);
+    this.updateQuantity = this.updateQuantity.bind(this);
+    this.resetState = this.resetState.bind(this);
+  }
+
+  updateMeasurement(e) {
+    this.setState({
+      measurementUnit: e.target.textContent
+    });
+  }
+
+  updateName(e) {
+    this.setState({
+      name: e.target.value
+    });
+  }
+
+  updateQuantity(e) {
+    this.setState({
+      quantity: e.target.value
+    });
+  }
+
+  categoryClicked(e) {
+    this.setState({
+      foodCategory: e.target.id
+    });
+  }
```

Comment:
If you add a `name` attribute to each input that matches the related property name in state, you can make a single `handleChange` method that'll work with all inputs.

```JSX
handleChange(e) {
  const name = e.target.name;
  const value = e.target.value;
  
  this.setState({
    [name]: value
  });
}
```